js/chat_app/src/components/DashBoard.js

```diff
@@ -27,6 +28,15 @@ export const DashBoard = () => {
     setOpen(false);
   };
 
+  useEffect(() => {
+    const url = `api/users/${auth?.user.id}/chats`;
+    const token = auth?.authToken;
+
+    getServerResponse(url, { "auth-token": token }).then((response) =>
```

Comment:
```
function useHttpClient() {
   const auth = useContext(AuthContext)

   return (url, {method, body}) => {
      return fetchJSON(url, {method, body, headers: { 'auth-token': auth.token }})
   }
}

function MyComponent() {
   const fetchJSON = useHttpClient()

   React.useEffect(() => {
      fetchJSON(url).then((response) => {
         console.log(response)
      })
   })
}
```

Reply:
<img width="634" alt="Screenshot 2022-11-03 at 12 30 42 PM" src="https://user-images.githubusercontent.com/43201911/199698655-1b2cc0a4-2d85-4eb0-b7ab-918f4027d813.png">
Do you mean something like that?